admin-console/src/main/webapp/WEB-INF/apsadmin/jsp/common/layouts/assets-more/portal/pageTree-extra.jsp

```diff
@@ -23,10 +41,152 @@
+                    } else {
+                        updateTree(data);
+                        alertService.addDismissableSuccess(response.actionMessages);
+                    }
+                    
+                    $(".alert-dismissable").fadeOut(8000);
+                    $('html, body').animate({ scrollTop: 0 }, 'fast');
+                }
+            });
+        }
+    
+        function expandNode (selector) {
+            $(selector).removeClass("hidden collapsed");
+            $('#pageTree .icon.fa-angle-right').removeClass('fa-angle-right')
+                .addClass('fa-angle-down');
+        }
+    
+        /**
+         * Refresh treegrid
+         */
+        function updateTree(treeNodes) {
+        	var previousParentId = $("#"+treeNodes.selectedNode).clone().data("parent");
+        	
+        	$("#"+treeNodes.parentPageCode).after($("#"+treeNodes.selectedNode)
+        		.attr("data-parent","#"+treeNodes.parentPageCode));
+    
+        	/* indent selectedNode */
+            indentNode(treeNodes.selectedNode, treeNodes.parentPageCode);
+            /* update selectedNode subTree */
+            sortSubtree(treeNodes.selectedNode);
+            
+            /* update previous parentNode */
+        	if($("[data-parent='"+previousParentId+"']").length == 0) {
+                $(previousParentId).find(".expand-icon").removeClass("fa-angle-down fa-angle-right");
+                $(previousParentId).find(".node-icon")
+                  .removeClass("fa-folder")
+                  .addClass("fa-folder-o");
+            }
+        	/* update current parentNode */
+        	if($("[data-parent='#"+treeNodes.parentPageCode+"']").length > 0) {
+                $("#"+treeNodes.parentPageCode).find(".expand-icon").addClass("fa-angle-down");
+                $("#"+treeNodes.parentPageCode).find(".node-icon")
+                  .addClass("fa-folder")
+                  .removeClass("fa-folder-o");
+        	}
+        }
+    
+        /**
+         * Update the treeNode's subTree element position recursively
+         */
+        function sortSubtree(treeNode) {
+        	var children = $("[data-parent='#"+treeNode+"']");
+            if(children.length > 0) {
+            	for(var i=0; i < children.length; i++) {
+                    $("#"+treeNode).after(children[i]);
+                    indentNode($(children[i]).attr("id"),treeNode);
+            		sortSubtree($(children[i]).attr("id"));
+            	}
+        	}
+        }
+    
+        /**
+         * Update the treeNode's indentation
+         */
+        function indentNode(treeNode, parentNode) {
+            var delta = $("#"+parentNode+" span.indent").length - $("#"+treeNode+" span.indent").length;
+            var indentNode = $("#"+treeNode);
+            if(delta >= 0) {
+                for(var i = 0; i <= Math.abs(delta); i++) {
+                    indentNode.find("span.indent").last().after('<span class="indent"></span>');
+                }
```

Comment:
this could probably be handled by a single span changing the css and creating a collection of indent classes to avoid adding and removing elements from the dom constantly:
i.e.
<span class="indent indent-size-2"></span>

just a comment anyway, it would require probably too much refactoring in too many places.